expression/integration_test.go

```diff
@@ -1768,10 +1768,15 @@ func (s *testIntegrationSuite) TestTimeBuiltin(c *C) {
 	result.Check(testkit.Rows("2017-01-01 2017-01-01 12:20:59 12:20:59"))
 	result = tk.MustQuery("select str_to_date('aaa01-01-2017', 'aaa%d-%m-%Y'), str_to_date('59:20:12 aaa01-01-2017', '%s:%i:%H aaa%d-%m-%Y'), str_to_date('59:20:12aaa', '%s:%i:%Haaa')")
 	result.Check(testkit.Rows("2017-01-01 2017-01-01 12:20:59 12:20:59"))
+	result = tk.MustQuery("select str_to_date('01-01-2017', '%d')")
+	result.Check(testkit.Rows("0000-00-01"))
+	tk.MustQuery("show warnings").Check(testutil.RowsWithSep("|", "Warning|1292|Truncated incorrect datetime value: '01-01-2017'"))
```

Comment:
In mysql5, it has two warnings.

```mysql
+---------+------+-----------------------------------------------------------------+
| Level   | Code | Message                                                         |
+---------+------+-----------------------------------------------------------------+
| Warning | 1292 | Truncated incorrect date value: '17-09-2010'                    |
| Warning | 1411 | Incorrect datetime value: '17-09-2010' for function str_to_date |
+---------+------+-----------------------------------------------------------------+
```

Reply:
I'm double checking this with the MySQL 5.6 warnings in implementation.

Reply:
Yes it is.

```
mysql> select str_to_date('01-01-2017', '%d');
Field   1:  `str_to_date('01-01-2017', '%d')`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       DATE
Collation:  binary (63)
Length:     10
Max_length: 0
Decimals:   0
Flags:      BINARY 


+---------------------------------+
| str_to_date('01-01-2017', '%d') |
+---------------------------------+
| NULL                            |
+---------------------------------+
1 row in set, 2 warnings (0.00 sec)

mysql> SHOW WARNINGS\G
*************************** 1. row ***************************
  Level: Warning
   Code: 1292
Message: Truncated incorrect date value: '01-01-2017'
*************************** 2. row ***************************
  Level: Warning
   Code: 1411
Message: Incorrect datetime value: '01-01-2017' for function str_to_date
2 rows in set (0.00 sec)
```

In MySQL 5.7, the test not only returns two warnings, but also returns NULL instead of `0000-00-01`.